agent/consul/state/catalog.go

```diff
@@ -1601,7 +1612,7 @@ func (s *Store) ensureCheckTxn(tx *txn, idx uint64, hc *structs.HealthCheck) err
 		existingCheck := existing.(*structs.HealthCheck)
 		hc.CreateIndex = existingCheck.CreateIndex
 		hc.ModifyIndex = existingCheck.ModifyIndex
-	} else {
+	} else if !preserveIndexes {
 		hc.CreateIndex = idx
 		hc.ModifyIndex = idx
```

Comment:
Is setting `hc.ModifyIndex` here necessary? If `existing == nil` I think we should be able to always make it down to L1691